onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorCol2Im.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "./precomp.h"
+
+namespace Dml
+{
+
+class DmlOperatorCol2Im : public DmlOperator, public Col2ImHelper
+{
+public:
+    explicit DmlOperatorCol2Im(const MLOperatorKernelCreationContext& kernelCreationContext)
+    :   DmlOperator(kernelCreationContext),
+        Col2ImHelper(kernelCreationContext, kernelCreationContext.GetTensorShapeDescription())
+    {
+        ML_CHECK_VALID_ARGUMENT(kernelCreationContext.GetInputCount() == 3, "Col2Im expects 3 inputs.");
+        ML_CHECK_VALID_ARGUMENT(kernelCreationContext.GetOutputCount() == 1, "Col2Im expects 1 output.");
+
+        auto tensorShapeDescription = kernelCreationContext.GetTensorShapeDescription();
+        std::vector<uint32_t> inputTensorShape = tensorShapeDescription.GetInputTensorShape(0);
+        std::vector<uint32_t> outputTensorShape = tensorShapeDescription.GetOutputTensorShape(0);
+
+        ML_CHECK_VALID_ARGUMENT(outputTensorShape.size() == m_outputShape.size());
+        for (uint32_t i = 0; i < outputTensorShape.size(); i++)
+        {
+            ML_CHECK_VALID_ARGUMENT(outputTensorShape[i] == m_outputShape[i]);
+        }
```

Comment:
```suggestion
        ML_CHECK_VALID_ARGUMENT(outputTensorShape == m_outputShape);
```
Are the two vectors not comparable?